otel agent resourcedetection processor compatibility docs for EKS (#23983)
newrelic/docs-website: src/content/docs/kubernetes-pixie/k8s-otel/advance-config.mdx

If your provider is GKE Autopilot, please do not change the value of `daemonset.containerSecurityContext.privileged` from the default because privileged containers are not supported by this provider.

> If your provider is EKS, please make either of the following updates to your cluster's terraform or in the AWS console to enable the `resourcedetection` processor:

Reviewer: ```suggestion
If you use Amazon EKS, update your cluster's Terraform configuration or use the AWS Console to enable the `resourcedetection` processor via one of the following methods:
```

>   * increase IMDS hop limit to 2 for node group(s) using `http_put_response_hop_limit`

Reviewer: ```suggestion
  * Increase the IMDS hop limit: Set the `http_put_response_hop_limit` to 2 for your node groups.
```

>   * enable pod identity for eks module using addon `eks-pod-identity-agent`

Reviewer: ```suggestion
  * Enable Pod Identity: Add the `eks-pod-identity-agent` add-on to your EKS module.
```
